group/assignment5/Assignment5.java

```diff
@@ -0,0 +1,149 @@
+
+                // insert firstWord's character if it isn't in the map and make a space for it in the graph; or just fetch it from map if it exists
+                Vertex firstWordsCharVertex = addOrFetchVertex(firstWordsCharacter);
+                //insert secondWrod's character if it isn't in the map (also works if the characters are the same, because it has just been inserted); or just fetch it from map if it exists
+                Vertex secondWordsCharVertex = addOrFetchVertex(secondWordsCharacter);
+
+                if (!firstWordsCharacter.equals(secondWordsCharacter) && checkForGraphEdges) {
+                    // different characters, both inserted in the map if needed
+                    // if they are the first two different characters discovered, 
+                    // insert edge in graph from firstWordsCharacter to secondWordsCharacter
+                    checkForGraphEdges = false;
+                    graphOfLetters.addEdge(firstWordsCharVertex, secondWordsCharVertex);
+                }
+                index++;
+            }
+
+            int secondWordIndex = index;
+
+            // process until the end of the first word if it is longer
+            while (index < firstWordLength) {
+                Character firstWordsCharacter = firstWord.charAt(index);
+                addOrFetchVertex(firstWordsCharacter);
+                index++;
+            }
+
+            // need to process the last word in dictionary, if it is longer
+            if (isSecondWordLastInDictionary(dictionaryList.size(), i + 1)) {
+                while (secondWordIndex < secondWordLength) {
+                    Character secondWordsCharacter = secondWord.charAt(secondWordIndex);
+                    addOrFetchVertex(secondWordsCharacter);
+                    secondWordIndex++;
+                }
+            }
+
+        }
+    }
+
+    private Vertex addOrFetchVertex(Character letter) {
+        Vertex vertex;
+        if (!graphOfLetters.containsVertex(letter)) {
+            vertex = graphOfLetters.addVertex(letter);
+        } else {
+            vertex = graphOfLetters.getVertex(letter);
+        }
+        return vertex;
+    }
+
+    private boolean isSecondWordLastInDictionary(int dictionarySize, int secondWordPosition) {
+        return secondWordPosition == (dictionarySize - 1);
+    }
+
+    public static void main(String[] args) {
+        LinkedList<String> dictionary = new LinkedList<>();
+        /* 
+        dictionary.add("ART");
+        dictionary.add("RAT");
+        dictionary.add("CAT");
+        dictionary.add("CAR");
+         */
+
+ /* 
+        dictionary.add("54");
+        dictionary.add("50");
+        dictionary.add("24");
+        dictionary.add("21");
+        dictionary.add("35");
+        dictionary.add("30");
+        dictionary.add("15");
+         */
+ 
+        //dictionary.add("ACE");
+        //dictionary.add("ALU");
+        dictionary.add("BEAR");

```

Comment:
You might consider deleting the commented code if you are not using it, as it is a bit confusing to leave it here.